### ISSUE TITLE
[JENKINS-6242] Pass through Matrix build values.

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/copyartifact-plugin#287]
<p>Although the plugin can copy artifacts from a specified axis without problem, it doesnt seem to be able to substitute parameters from a matrix project to select the appropiate axis from a sub job.</p>

<p>i.e.</p>

<p>Job1 is a matrix project that builds shared objects for a variety of platforms e.g. linux/windows/mac</p>

<p>Job2 is a matrix project that builds an executable that depends on Job1 shared object, copyartifact should only copy the axis that is relevant. e.g Job1/Platform=$Platform</p>

<p>There are workarounds for this, but they get very messy and involve copying every combination over.  Is this possible?  Is it already possible and I missed it?</p>

<p>Many Thanks,<br/>
Paul.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=pdwakefield">pdwakefield</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-6242" target="_blank">Pass through Matrix build values.</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: copyartifact-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2010-04-14T10:46:33+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 20260107-133739
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Although the plugin can copy artifacts from a specified axis without problem, it doesnt seem to be able to substitute parameters from a matrix project to select the appropiate axis from a sub job.</p>

<p>i.e.</p>

<p>Job1 is a matrix project that builds shared objects for a variety of platforms e.g. linux/windows/mac</p>

<p>Job2 is a matrix project that builds an executable that depends on Job1 shared object, copyartifact should only copy the axis that is relevant. e.g Job1/Platform=$Platform</p>

<p>There are workarounds for this, but they get very messy and involve copying every combination over.  Is this possible?  Is it already possible and I missed it?</p>

<p>Many Thanks,
Paul.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>NA</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-6242?focusedId=137810&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-137810">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>As you might guess, I never tried my plugin with a matrix project at all.. I was pleased to find it won't take much to make this work.</p>

<p>1) some text in the help about using matrix projects.. turns out you can enter something like "my-matrix-job/jdk=1.5" to get a particular configuration of a matrix project (sounds like you figured that out).
2) expand variables in the project name as already done for artifacts to copy and target path.. THEN you'll be able to make a new matrix project with axes matching some other job, and use copyartifact with source project = other-matrix-job/jdk=$jdk</p>
</pre>
</details>

</ul>
</details>

<p>As you might guess, I never tried my plugin with a matrix project at all.. I was pleased to find it won't take much to make this work.</p>

<p>1) some text in the help about using matrix projects.. turns out you can enter something like "my-matrix-job/jdk=1.5" to get a particular configuration of a matrix project (sounds like you figured that out).<br/>
2) expand variables in the project name as already done for artifacts to copy and target path.. THEN you'll be able to make a new matrix project with axes matching some other job, and use copyartifact with source project = other-matrix-job/jdk=$jdk</p>
